support/xspec-dev/xspec-mx-html-report.xsl

```diff
@@ -0,0 +1,490 @@
+   </xsl:template>
+   
+   <xsl:template priority="20" match="x:test/@successful"/>
+   
+   
+   <xsl:template match="x:scenario">
+      <details class="{ (@pending/'pending',.[descendant::x:test/@successful = 'false']/'failing','passing')[1]}{ child::x:test[1]/' tested' }">
+         <xsl:copy-of select="@id"/>
+         <xsl:if test="descendant::x:test/@successful = 'false'">
+            <xsl:attribute name="open">open</xsl:attribute>
+         </xsl:if>
+         <summary>
+            <xsl:apply-templates select="." mode="in-summary"/>
+         </summary>
+         <div class="{ (child::x:scenario[1]/'scenario-results','test-results')[1] }">
+           <xsl:apply-templates/>
+         </div>
+      </details>
+   </xsl:template>
+   
+   <xsl:template match="x:scenario/x:label"/>
+   
+   <xsl:template match="x:report | x:scenario" mode="in-summary">
+      <xsl:variable name="here" select="."/>
+      <xsl:variable name="subtotal" select="exists(ancestor-or-self::x:report//x:test except descendant::test)"/>
+      <xsl:variable name="all-passing" select="empty(descendant::x:test[not(@successful='true')])"/>
+      <span class="label">{ self::x:report/'Total tests '}{ child::x:label }</span>
+      <xsl:text>&#xA0;</xsl:text><!-- nbsp -->
+      <xsl:apply-templates select="self::x:scenario/descendant::x:test" mode="icon"/>
+      <span class="result">
+      <xsl:iterate select="'passing', 'pending', 'failing','total'[not($subtotal)],'subtotal'[$subtotal]">
+         
+         <xsl:variable name="this-count">
+            <xsl:apply-templates select="." mode="count-tests">
+               <xsl:with-param name="where" select="$here"/>
+            </xsl:apply-templates>
+         </xsl:variable>
+         <xsl:text> </xsl:text>
+         <span class="{ . }{ ' zero'[$this-count='0']}{ .[.=('total','subtotal')] ! ' passing'[$all-passing]}">
+            <b class="pg">{ . }</b>
+            <xsl:text>: </xsl:text>
+            <xsl:sequence select="$this-count"/>
+         </span>
+      </xsl:iterate>
+      </span>
+   </xsl:template>
+   
+   <xsl:template priority="20" match="x:scenario[exists(child::x:test)]" mode="in-summary">
+      <xsl:variable name="here" select="."/>
+      <span class="label">{ child::x:label }</span>
+      <xsl:text>&#xA0;</xsl:text><!-- nbsp -->
+      <xsl:apply-templates select="child::x:test" mode="icon"/>
+      <xsl:apply-templates select="child::x:test" mode="#current"/>
+      <span class="result { child::x:test/(@pending/'pending', @successful)[1] }">{ (child::x:test/@pending/('Pending ' || .),child::x:test[@successful='true']/'Passes','Fails')[1] }</span>
+   </xsl:template>
+   
+   <xsl:template mode="in-summary" match="x:test">
+      <xsl:text> </xsl:text>
+      <span class="testing"> Expecting <span class="expecting">{ child::x:label }</span></span>
```

Comment:
It's a matter of preference, but I found myself wishing for sibling x:expect items to appear in the report on separate lines instead of running together.